Implemented BaseScrollView and added to ProfileVC.

kforkarim opened the pull request:
Implemented BaseScrollView and added to ProfileVC.

compnerd committed:
swiftlint: ignore closure parameters, TODO

kforkarim committed:
Implemented BaseScrollView and added to ProfileVC.

compnerd commented on WalkingChallenge/BaseScrollView.swift:
```diff
@@ -0,0 +1,33 @@
+//
+//  Created by Karim Abdul on 4/18/17.
+//  Copyright © 2017 AKDN. All rights reserved.
+//
```
Please use the standard header in the rest of the files.

compnerd commented on WalkingChallenge/BaseScrollView.swift:
```diff
@@ -0,0 +1,33 @@
+  }
+  required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+    confireView()
```
ITYM `configureView`

compnerd commented on WalkingChallenge/BaseScrollView.swift:
```diff
@@ -0,0 +1,33 @@
+  let contentView = UIView()
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    confireView()
```
ITYM `configureView`

compnerd commented on WalkingChallenge/BaseScrollView.swift:
```diff
@@ -0,0 +1,33 @@
+      make.bottom.equalToSuperview()
+      make.leading.equalToSuperview()
+      make.trailing.equalToSuperview()
+      make.width.equalToSuperview()
```
This can be written more succinctly as:

    make.edges.equalToSuperview()

compnerd replied:
Isn't the `make.width.equalToSuperview()` now unnecesary?

compnerd commented on WalkingChallenge/ProfileViewController.swift:
```diff
@@ -264,6 +265,13 @@ class ProfileViewController: UIViewController {
+      make.top.equalToSuperview()
+      make.bottom.equalToSuperview()
+      make.leading.equalToSuperview()
+      make.trailing.equalToSuperview()
```
This can be written more succinctly as:

    make.edges.equalToSuperview()

kforkarim committed:
Updated constraints to edges.

compnerd commented on WalkingChallenge/ProfileViewController.swift:
```diff
@@ -235,7 +235,7 @@ class ProfileViewController: UIViewController {
     configureView()
   }
-
+  
```
bleeding whitespace

kforkarim committed:
Fixed ScrollView issue.

kforkarim committed:
Merge branch 'master' into scrollView_Implementation

compnerd committed:
Xcode: add group

compnerd committed:
split out EventDataSource

compnerd committed:
Xcode: add group

compnerd committed:
Event: split out model

compnerd committed:
ProfileView: split out configuration of the view

Refactor the supporter and events view into separate functions.  This
simplifies the configureView function, allows us to make it more dynamic
and to properly handle the rendering of the supporters and events in the
range [0,n].  It also opens up the possibility for refactoring further
into a custom container view to simplify the layout constraints.

compnerd committed:
lint: clean up some whitespace

compnerd committed:
ProfileView: split configureView further

compnerd committed:
EventsView: add empty events view

compnerd committed:
EventsView: mockup events list

compnerd committed:
ConfigurationView: mockup buttons

compnerd committed:
ConfigurationView: extract disclosure indicator buttons

Add a helper to create a disclosure indicator button which we use in a
few places in the UI.  This allows us to create similar buttons
throughout the interface.

compnerd committed:
SupportersView: scale sponsors by 2

compnerd committed:
ProfileView: partially mock up statistics

compnerd committed:
lint: make delegates weak

compnerd committed:
TeamView: style

compnerd committed:
TeamView: refactor view layout

compnerd committed:
TeamView: partially mock up statistics

compnerd committed:
TeamView: minor cleanup of view controller display

compnerd committed:
fill in image placeholders

compnerd committed:
stifle a few compiler warnings

kforkarim committed:
ScollView Implemented.

compnerd commented on WalkingChallenge/ProfileViewController.swift:
```diff
@@ -390,21 +391,26 @@ class ProfileViewController: UIViewController {
+    }
     view.backgroundColor = Style.Colors.white
-
+    
```
Bleeding whitespace

compnerd commented on WalkingChallenge/ProfileViewController.swift:
```diff
@@ -390,21 +391,26 @@ class ProfileViewController: UIViewController {
     view.backgroundColor = Style.Colors.white
-
+    
     var top: ConstraintRelatableTarget = topLayoutGuide.snp.bottom
```
Change this to `scrollView.snp.top`

compnerd commented on WalkingChallenge/ProfileViewController.swift:
```diff
@@ -242,26 +243,26 @@ class ProfileViewController: UIViewController {
     profileImage.layer.masksToBounds = true
     profileImage.snp.makeConstraints { (make) in
-      make.top.equalTo(top).offset(Style.Padding.p12)
+      make.top.equalToSuperview().offset(Style.Padding.p12)
```
Changing the initialization of the parameter below should prevent this bit of the change.
